src/compiler/scala/tools/nsc/settings/Warnings.scala

```diff
@@ -17,6 +17,7 @@ trait Warnings {
 
   // Non-lint warnings.
 
+  val warnMacros           = BooleanSetting("-Ywarn-macros", "Enable warnings on macro expansions.")
```

Comment:
I didn't give a lot of thought to this. For this ticket, I think linting looks at both. The other use case is the ticket whether to emit warnings on unexpanded code. So maybe the space of options is none, both, unexpanded and expanded.